src/globals.c

```diff
@@ -318,25 +318,47 @@ bool init_globals(void)
 		return false;
 	}
 
-/* Set configuration defaults based on options provided
-	at configure time.  */
+/* Set defaults with following order of preference:
+	1. Runtime flags
+	2. State dir configuration files
+	3. Configure time settings
+*/
+	if (!set_format_string(NULL)) {
 #ifdef FORMATID
-	set_format_string(FORMATID);
-#else
-	(void)set_format_string(NULL);
-#endif /* FORMATID */
+		/* Fallback to configure time format_string if other sources fail */
+		set_format_string(FORMATID);
```

Comment:
Perhaps we should add error handling here, and move the `printf("\nDefault format ...)` message to this function instead?